### PR TITLE
CODENVY-232: add element injector service by parent id

### PR DESCRIPTION
**Pull request by olexii4**
**_2 Upvotes**_ Signed-off-by: Oleksii Orel oorel@codenvy.com

### CODE REVIEW

File: dashboard/src/components/injector/che-injector.service.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+
+/**
+ * This class is handling the service for the additional element injection.
```
Comment (ashumilova) [inline dashboard/src/components/injector/che-injector.service.js]: This class is handling the service for the injecting UI elements to other widgets from outside

File: dashboard/src/components/injector/che-injector.service.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+   * @ngInject for Dependency injection
+   */
+  constructor($timeout, $document, $compile) {
+    this.$timeout = $timeout;
+    this.$document = $document;
+    this.$compile = $compile;
+
+    //object of destination elements with id as keys and innerHTML of additional widget as value
+    //{ parentElementId: { additionalElementId: additionalElementInnerHTML } }
+    this.additionalElements = {};
```
Comment (ashumilova) [inline dashboard/src/components/injector/che-injector.service.js]: elementsToAdd, elementsToInject or simply elements

File: dashboard/src/components/injector/che-injector.service.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+
+/**
+ * This class is handling the service for the additional element injection.
+ * @author Oleksii Orel
+ */
+export class CheInjectorService {
```
Comment (ashumilova) [inline dashboard/src/components/injector/che-injector.service.js]: CheInjectorService = CheUIElementsInjectorService

File: dashboard/src/components/injector/che-injector.service.js (hunk 2 above)
Comment (ashumilova) [inline dashboard/src/components/injector/che-injector.service.js]: a Map can be used here

File: dashboard/src/components/injector/che-injector.service.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+          // set an id into the additional element
+          additionalElement.attr('id', additionalElementId);
+          this.injectAdditionalElement(parentElement, additionalElement);
+        }
+      }
+    });
+  }
+
+  /**
+   *  Inject an additional elements if it possible.
```
Comment (ashumilova) [inline dashboard/src/components/injector/che-injector.service.js]: Inject an additional elements if it possible. - one element

File: dashboard/src/components/injector/che-injector.service.js (hunk 3 above)
Comment (olexii4) [inline dashboard/src/components/injector/che-injector.service.js]: totally agree

File: dashboard/src/components/injector/che-injector.service.js (hunk 2 above)
Comment (olexii4) [inline dashboard/src/components/injector/che-injector.service.js]: elementsToInject  
To me, it is understandable.
Done

In this case I prefer work with the object

File: dashboard/src/components/injector/che-injector.service.js (hunk 1 above)
Comment (olexii4) [inline dashboard/src/components/injector/che-injector.service.js]: Done

File: dashboard/src/components/injector/che-injector.service.js (hunk 4 above)
Comment (olexii4) [inline dashboard/src/components/injector/che-injector.service.js]: typo
Done

**Comment by ashumilova** [comment]
+1

**Comment by akurinnoy** [comment]
+1

**Pull request merged**